runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -47,11 +47,12 @@
 // the performance difference in a playground just to see if it's worth
 // investigating further.
 
-// TODO put these into a struct?
-static void*  heap_base = NULL;
-static size_t heap_size = 0;
-static size_t cur_heap_offset = 0;
-static pthread_mutex_t chunk_alloc_lock;
+static struct shared_heap {
+  void* base;
+  size_t size;
+  size_t cur_offset;
+  pthread_mutex_t alloc_lock;
+} heap; // static, will be "zero" initialized automatically
```

Comment:
I might have been unclear -- I brought up the PTHREAD_MUTEX_INITIALIZER only to say that you could write a real initializer and wouldn't have to do the artificial-looking /* default init mutex */ thingie.  Given the requirement for making the pthread_mutex_destroy() call I like having the pthread_mutex_init() call for symmetry, which is why I said that having the static init would also be superfluous/redundant.  Or if you want to just leave the initializer off entirely that's also fine with me.